### PR TITLE
DB data storage and retrieval for the DTO + DAO's

### PR DESCRIPTION
**Pull request by Bsabz**
Intend to fix: #71 #75

### CODE REVIEW

File: src/main/java/www/purple/mixxy/dao/ComicDao.java [hunk 1]
```diff
@@ -33,15 +33,11 @@ public ComicsDto getAllComics() {
...
     }
     
-    public List<Comic> getComics(String author) {
-      return Collections.EMPTY_LIST;
-    }
-    
-    public List<Comic> getComics(long author) {
-      return Collections.EMPTY_LIST;
+    public List<Comic> getComics(String userName) {
+    	return objectify.get().load().type(Comic.class).filter("username", userName).list();
```
Comment (JesseTG) [inline src/main/java/www/purple/mixxy/dao/ComicDao.java]: Mixed tabs and spaces make me sad.  :(

File: src/test/java/www/purple/mixxy/controllers/ComicTest.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import www.purple.mixxy.models.Comic;
+import www.purple.mixxy.models.User;
+
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+@SuppressWarnings("PMD.JUnitTestContainsTooManyAsserts")
+public class ComicTest extends NinjaAppengineBackendTest {
```
Comment (JesseTG) [inline src/test/java/www/purple/mixxy/controllers/ComicTest.java]: Refresh my memory, was there some failure that you were unsure about?

File: src/main/java/www/purple/mixxy/dao/ComicDao.java (hunk 1 above)
Comment (Bsabz) [inline src/main/java/www/purple/mixxy/dao/ComicDao.java]: Fixed tabs + spacing

File: src/test/java/www/purple/mixxy/controllers/ComicTest.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+		ofy.save().entity(comic).now();
+
+		// Test that the post has been created
+		assertNotNull(ofy.load().type(Comic.class).first().now());
+
+		// Retrieve all posts created by Bob
+		List<Comic> bobPosts = ofy.load().type(Comic.class).filter("author", yetAnotherBob).list();
+
+		// Tests
+		assertEquals(1, bobPosts.size());
```
Comment (Bsabz) [inline src/test/java/www/purple/mixxy/controllers/ComicTest.java]: @JesseTG, Apparently I get size 0 here when trying to retrieve the comic posted. but apparently the comic saves in the comic entity.

Comment (JesseTG) [inline src/test/java/www/purple/mixxy/controllers/ComicTest.java]: Doesn't look like the `Comic` is associated with Bob in any way.  [See here.](https://github.com/objectify/objectify/wiki/Entities#refs)  Make `comic.author = Ref.create(yetAnotherBob)` before you save it.

**Comment by Bsabz** [comment]
This possibly solved #105

**Comment by JesseTG** [comment]
Merge, but keep an eye on the indices in the data store.

**Pull request merged**